std/src/main/scala/algebra/std/map.scala

```diff
@@ -130,3 +133,25 @@ class MapRng[K, V](implicit V: Rng[V]) extends MapSemiring[K, V] with Rng[Map[K,
   override def minus(x: Map[K, V], y: Map[K, V]): Map[K, V] =
     subtractMap(x, y)(V.minus)(V.negate)
 }
+
+class MapGenBool[K, V](implicit GBV: GenBool[V], EqV: Eq[V]) extends GenBool[Map[K, V]] {
```

Comment:
I consider `k -> 0` equivalent, for all purposes, to not having a mapping for key `k` at all. So I don't really care whether zeros are kept or not, except for maybe saving some space.

If we distinguish between `k -> 0` and "no value for key `k`", the behavior is unlawful. (I was actually wondering how is it possible that the tests for e.g. `MapRng` don't fail and then I noticed the `Eq` instance is smart enough to ignore zero values.)

In this setting, the user should only ever call `getOrElse(k, zero)` and then there will be no surprises.

I can imagine a case where it is tempting to encode some additional information in `k -> 0` vs. "no value for `k`", but I'm not sure it is a good idea in combination with `Rng` or `GenBool` operations.

Also note that an explicit call to remove zeros is not possible in a generic setting, where you only have a type class instance operating on some generic values.